dags/postgres_connection_example.py

```diff
@@ -1,57 +1,60 @@
 import boto3
+import logging
 import psycopg2
-import sshtunnel
 
 from airflow import DAG
 from airflow.models import Variable
+from airflow.models.connection import Connection
 from airflow.operators.python import PythonOperator
+from airflow.providers.common.sql.operators.sql import SQLExecuteQueryOperator
 
 
 def get_iam_token(host, port, user, region):
     client = boto3.client("rds")
-    token = client.generate_db_auth_token(DBHostname=host, Port=port, DBUsername=user, Region=region)
+    logging.info(user)
+    token = client.generate_db_auth_token(
+        DBHostname=host, Port=port, DBUsername=user, Region=region
+    )
     return token
 
 
 def connect_via_ssh():
-    ssh_host = Variable.get("SSH_HOST")
-    ssh_port = 22
-    ssh_user = Variable.get("SSH_USER")
-    ssh_pkey = '/usr/local/airflow/files/ssh_host.pem'  # Path to your SSH private key file
-    db_host = Variable.get("POSTGRES_DB_HOST")
-    db_user = Variable.get("POSTGRES_DB_USER")  # This is the IAM user or role
-    db_name = Variable.get("POSTGRES_DB_NAME")
+    db_host = Variable.get("RDS_HOST")
+    db_user = Variable.get("RDS_USER")  # This is the IAM user or role
     db_port = 5432
-    region = 'us-west-2'
-
-    with sshtunnel.SSHTunnelForwarder(
-            (ssh_host, ssh_port),
-            ssh_username=ssh_user,
-            ssh_pkey=ssh_pkey,
-            remote_bind_address=(db_host, db_port)
-    ) as tunnel:
-        token = get_iam_token(db_host, db_port, db_user, region)
-        connection = psycopg2.connect(
-            user=db_user,
-            password=token,
-            host='127.0.0.1',
-            port=tunnel.local_bind_port,
-            database=db_name,
-            sslmode='require'
-        )
-        cursor = connection.cursor()
-        cursor.execute("select * from information_schema.tables limit 5;")
-        result = cursor.fetchall()
-        print(result)
-        cursor.close()
-        connection.close()
+    region = "us-west-2"
+
+    token = get_iam_token(db_host, db_port, db_user, region)
+    """Establish a connection to a postgres database."""
+    conn = Connection.get_connection_from_secrets("test_postgres_iam_conn")
+    conn_args = {
+        "host": conn.host,
+        "user": conn.login,
+        "password": token,
+        "dbname": conn.schema,
+        "port": conn.port,
+    }
+    connection = psycopg2.connect(**conn_args)
+    cursor = connection.cursor()
+    cursor.execute("select * from information_schema.tables limit 5;")
+    result = cursor.fetchall()
+    print(result)
+    cursor.close()
+    connection.close()
```

Comment:
Is it worth keeping this Python task since we now have an SSH tunnel?

If no, then I think you can remove this.

If yes, then maybe you can comment on the pros and cons of this approach versus lines 56-60.

Reply:
Added comments

Reply:
The comments here help. I think it would also be good to rename the task IDs. I had trouble identifying which one does what when I was running them in the UI.